src/icons/Status.js

```diff
@@ -168,27 +168,25 @@ Loading.propTypes = {
     className: propTypes.string,
 }
 
-export const StatusIconNoDefault = ({
-    error,
-    warning,
-    valid,
-    loading,
-    className,
-}) =>
-    valid ? (
-        <Valid className={className} />
-    ) : warning ? (
-        <Warning className={className} />
-    ) : error ? (
-        <Error className={className} />
-    ) : loading ? (
-        <Loading className={className} />
-    ) : null
-
-StatusIconNoDefault.propTypes = {
-    valid: propTypes.bool,
-    error: propTypes.bool,
-    warning: propTypes.bool,
-    loading: propTypes.bool,
+export const StatusIcon = ({ status, defaultTo: DefaultTo, className }) => {
+    switch (status) {
+        case 'info':
+            return <Info className={className} />
+        case 'warning':
+            return <Warning className={className} />
+        case 'loading':
+            return <Loading className={className} />
+        case 'error':
+            return <Error className={className} />
+        case 'valid':
+            return <Valid className={className} />
+        default:
+            return DefaultTo ? <DefaultTo className={className} /> : null
```

Comment:
It would be more consistent with the rest of the codebase if we use this prop like this:
```
<StatusIcon
    defaultTo={<MyDefaultIcon className="yeah-baby />}
/>
```
And also, I think we should not default to null in the switch statement, but rather use a `defaultProp`.

The default case would then look like this:
`return defaultTo`

And you also wouldn't need to alias the prop in the function signature.

FYI: we are using this same approach approach for icons in components such as `AlertBar`, `Chip`, `Tab` etc....

Reply:
Ok n.p. In a review @Mohammer5 warned me against using defaultProps too much, as we didn't use them too much in ui-core, but that might have been a special case. That's the reason I'm not using it here. I'm fine with your suggestion though, seems good to me 👍.